src/server/editorServices.ts

```diff
@@ -931,30 +931,28 @@ namespace ts.server {
                 return;
             }
 
-            this.logger.startGroup();
-
-            let counter = 0;
-            counter = printProjects(this.logger, this.externalProjects, counter);
-            counter = printProjects(this.logger, this.configuredProjects, counter);
-            counter = printProjects(this.logger, this.inferredProjects, counter);
-
-            this.logger.info("Open files: ");
-            for (const rootFile of this.openFiles) {
-                this.logger.info(rootFile.fileName);
-            }
-
-            this.logger.endGroup();
-
-            function printProjects(logger: Logger, projects: Project[], counter: number) {
-                for (const project of projects) {
-                    project.updateGraph();
-                    logger.info(`Project '${project.getProjectName()}' (${ProjectKind[project.projectKind]}) ${counter}`);
-                    logger.info(project.filesToString());
-                    logger.info("-----------------------------------------------");
-                    counter++;
+            this.logger.group(info => {
+                let counter = 0;
+                printProjects(this.externalProjects);
+                printProjects(this.configuredProjects);
+                printProjects(this.inferredProjects);
+
+                info("Open files: ");
+                for (const rootFile of this.openFiles) {
+                    info(rootFile.fileName);
+                }
+
+                function printProjects(projects: Project[]) {
+                    for (const project of projects) {
+                        project.updateGraph();
+                        info(`Project '${project.getProjectName()}' (${ProjectKind[project.projectKind]}) ${counter}`);
+                        info(project.filesToString());
+                        info("-----------------------------------------------");
+                        counter++;

```

Comment:
@sandersn, `counter` is used on line [948](https://github.com/Microsoft/TypeScript/pull/17271/files#diff-efa731ca7c6279501f903db87403d2beR948), though its still feasible to do something like:

```ts
function printProjects(projects: Project[], counter: number) {
  for (const project of projects) {
    ...
    counter++;
  }
  return counter;
}
```

And then use it like this:

```ts
let counter = 0;
counter = printProjects(this.externalProjects, counter);
counter = printProjects(this.configuredProjects, counter);
printProjects(this.inferredProjects, counter);
```